lib/jnpr/junos/decorators.py

```diff
@@ -85,36 +85,54 @@ def ignoreWarnDecorator(function):
+                                    'statement not found'])
         cu.load(cnf, ignore_warning='statement not found')
 
-    :ignore_warning: It can take take boolean value or string or list of string.
-        if True, it will ignore all warning. If string, it will ignore warning if
-        the statement matches given string. If list of strings, it will try to check
-        if warning statement is from any of the given strings in the list:
+    :ignore_warning: It can take take boolean value or string or list of
+        string. If True, it will ignore all warning. If string, it will
+        ignore warning if the statement matches given string. If list of
+        strings, it will try to check if warning statement is from any of the
+        given strings in the list.
+
+    .. note::
+            When the value of ignore_warning is a string, or list of strings,
+            the string is actually used as a case-insensitive regular
+            expression pattern. If the string contains only alpha-numeric
+            characters, as shown in the above examples, this results in a
+            case-insensitive substring match. However, any regular expression
+            pattern supported by the re library may be used for more
+            complicated match conditions.
     """
     @wraps(function)
     def wrapper(*args, **kwargs):
-        if 'ignore_warning' in kwargs:
-            ignore_warn = kwargs.pop('ignore_warning')
+        ignore_warn = kwargs.pop('ignore_warning', False)
+        if ignore_warn:
             try:
                 result = function(*args, **kwargs)
                 return result
             except RpcError as ex:
                 ex.rpc_xml = JXML.remove_namespaces(ex.rpc_xml)
-                if hasattr(ex, 'rpc_error') and\
-                        ex.rpc_error['severity'] == 'warning':
-                    if ignore_warn is True:
-                        return ex.rpc_xml
-                    elif isinstance(ignore_warn, (str, unicode)):
-                        if re.search(ignore_warn, ex.message, re.I):
-                            return ex.rpc_xml
-                    elif isinstance(ignore_warn, list):
-                        for warn_msg in ignore_warn:
-                            if re.search(warn_msg, ex.message, re.I):
-                                return ex.rpc_xml
-                    raise ex
+                if hasattr(ex, 'errs'):
+                    for err in ex.errs:
+                        if err['severity'] == 'warning':
+                            if isinstance(ignore_warn, (str, unicode)):
+                                if not re.search(ignore_warn, err['message'],
+                                                 re.I):
+                                    raise ex
+                            elif isinstance(ignore_warn, list):
+                                for warn_msg in ignore_warn:
+                                    if not re.search(warn_msg, err['message'],
```

Comment:
I think this logic is wrong. say xml reply is

```
<rpc-error>
        <error-severity>warning</error-severity>
        <source-daemon>l2cpd</source-daemon>
        <error-path>[edit protocols]</error-path>
        <error-info>
            <bad-element>lldp</bad-element>
        </error-info>
        <error-message>Advertisement-interval is less than four</error-message>
    </rpc-error>
    <rpc-error>
        <error-severity>warning</error-severity>
        <source-daemon>chassisd</source-daemon>
        <error-path>[edit]</error-path>
        <error-info>
            <bad-element>chassis</bad-element>
        </error-info>
        <error-message>A system reboot is mandatory</error-message>
    </rpc-error>
```
errs is
```python
[{'severity': 'warning', 'edit_path': None, 'message': 'Advertisement-interval is less than four', 'bad_element':'l2cpd', 'source': ‘l2cpd’},
 {'severity': 'warning', 'edit_path': None, 'message': 'A system reboot is mandatory', 'bad_element': 'chassis', 'source': ‘chassisd’}]
```

if I give ignore_warning in different order
ignore_warning = ['A system reboot is mandatory’, 'Advertisement-interval is less than four']

'A system reboot is mandatory’ will be matched with 'Advertisement-interval is less than four' resulting in failure. Hence raising exception. 

**Note**: I have not tested this.